pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,267 @@
+            )
+        n_cycles = iter(n_cycles)
+    return n_cycles
+
+
+def _create_freqs(freq_start, freq_stop, freq_step=1):
+    """
+    Creates an array of frequencies.
+
+    ..todo:: Implement log scaling
+
+    Parameters
+    ----------
+    freq_start : float
+        Starting value for the frequency definition.
+    freq_stop: float
+        Stopping value for the frequency definition, inclusive.
+    freq_step: float, optional
+        Step value, for linearly spaced values between start and stop.
+
+    Returns
+    -------
+    freqs: 1d array
+        Frequency indices.
+    """
+    return np.arange(freq_start, freq_stop + freq_step, freq_step)
+
+
+def compute_wavelet_transform(sig, fs, freqs, n_cycles=1.5, scaling=1.0, norm="amp"):
+    """
+    Compute the time-frequency representation of a signal using morlet wavelets.
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float or None
+        Sampling rate, in Hz.
+    freqs : 1d array or list of float
+        If array, frequency values to estimate with morlet wavelets.
+        If list, define the frequency range, as [freq_start, freq_stop, freq_step].
+        The `freq_step` is optional, and defaults to 1. Range is inclusive of `freq_stop` value.
+    n_cycles : float or 1d array
+        Length of the filter, as the number of cycles for each frequency.
+        If 1d array, this defines n_cycles for each frequency.
+    scaling : float
+        Scaling factor.
+    norm : {'sss', 'amp'}, optional
+        Normalization method:
+
+        * 'sss' - divide by the square root of the sum of squares
+        * 'amp' - divide by the sum of amplitudes
+
+    Returns
+    -------
+    mwt : 2d array
+        Time frequency representation of the input signal.
+
+    Notes
+    -----
+    This computes the continuous wavelet transform at specified frequencies across time.
+    """
+    if not isinstance(sig, (nap.Tsd, nap.TsdFrame, nap.TsdTensor)):
+        raise TypeError("`sig` must be instance of Tsd, TsdFrame, or TsdTensor")
+    if isinstance(freqs, (tuple, list)):
+        freqs = _create_freqs(*freqs)
+    if fs is None:
+        fs = sig.index.shape[0] / (sig.index.max() - sig.index.min())
+    n_cycles = _check_n_cycles(n_cycles, len(freqs))
+    if isinstance(sig, nap.Tsd):
+        sig = sig.reshape((sig.shape[0], 1))
+        output_shape = (sig.shape[0], len(freqs))
+    else:
+        output_shape = (sig.shape[0], len(freqs), *sig.shape[1:])
+        sig = sig.reshape((sig.shape[0], np.prod(sig.shape[1:])))
+    mwt = np.zeros(
+        [sig.values.shape[0], len(freqs), sig.values.shape[1]], dtype=complex
+    )
+    for channel_i in range(sig.values.shape[1]):
+        for ind, (freq, n_cycle) in enumerate(zip(freqs, n_cycles)):
+            mwt[:, ind, channel_i] = _convolve_wavelet(
+                sig[:, channel_i], fs, freq, n_cycle, scaling, norm=norm
+            )
```

Comment:
if the signal is a `TsdFrame`, you don't need to loop, you can call 

nap.convolve(sig, wavelet_kernel[0, i]), nap.convolve(sig, wavelet_kernel[1,i])

this handles the epochs and works with pynajax. the loop you'll need is on the filter bank.

I would argue that we should change `nap.convolve` to handle ND arrays so that the loop is done at the level of the convolve function instead, and here no loops are needed.